### PR TITLE
[Hieu] Add chat

### PR DESCRIPTION
**Pull request by HieuJC**

### CODE REVIEW

File: app/src/main/java/com/hieuminh/chatapp/enum/RTFirebaseAttributeName.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.hieuminh.chatapp.enum
+
+enum class RTFirebaseAttributeName(val index: String) {
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/enum/RTFirebaseAttributeName.kt]: .

File: app/src/main/java/com/hieuminh/chatapp/presenter/impl/AddChatPresenterImpl.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                }
+            }
+        }
+    }
+
+    override fun onDestroy() {
+        view = null
+    }
+
+    private fun addChatIdForUser(userId1: String, userId2: String, chatId: String) {
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/presenter/impl/AddChatPresenterImpl.kt]: what is userId1, userId2 ??

File: app/src/main/java/com/hieuminh/chatapp/ui/home/AddChatFragment.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        toolBar.setNavigationOnClickListener { mAddChatNavigator.returnHomeFragment() }
+
+        btSearchEmail.setOnClickListener {
+            mAddChatPresenter.search(
+                etAddChat.text.toString().trim()
+            )
+        }
+
+        btAddChat.setOnClickListener { mAddChatPresenter.addChat(userProfile) }
```
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/ui/home/AddChatFragment.kt]: extract to handleEvents()

File: app/src/main/java/com/hieuminh/chatapp/extensions/StringExtension.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.hieuminh.chatapp.extensions
+
+fun String.lastName(): String {
+    val list = this.split(" ")
+    return list.last()
+}
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/extensions/StringExtension.kt]: exist in string util

File: app/src/main/java/com/hieuminh/chatapp/ui/home/AddChatFragment.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    override fun onDestroy() {
+        mAddChatPresenter.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun onSearchSuccess(userProfile: UserProfileModel?) {
+        if (userProfile == null) {
+            llSearchFailed.visibility = VISIBLE
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/ui/home/AddChatFragment.kt]: create view extension,

File: app/src/main/res/layout/fragment_add_chat.xml [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+                <de.hdodenhof.circleimageview.CircleImageView
+                    android:id="@+id/civAvatar"
+                    android:layout_width="@dimen/dp_70"
+                    android:layout_height="@dimen/dp_70"
+                    android:src="@drawable/ic_baseline_person_24" />
+
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="70dp"
```
Comment (AnhJobChat) [inline app/src/main/res/layout/fragment_add_chat.xml]: .

**Review by vuphu — APPROVED** [review]

File: app/src/main/java/com/hieuminh/chatapp/model/ChatModel.kt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.hieuminh.chatapp.model
+
+class ChatModel() {
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/model/ChatModel.kt:3]: u can use data class for this case

File: app/src/main/java/com/hieuminh/chatapp/ui/MainActivity.kt [hunk 8]
```diff
@@ -10,4 +10,7 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
+
+    override fun onBackPressed() {
+    }
```
Comment (AnhJobChat) [inline app/src/main/java/com/hieuminh/chatapp/ui/MainActivity.kt]: remove or note todo

**Review by AnhJobChat — APPROVED** [review]

File: app/src/main/java/com/hieuminh/chatapp/ui/MainActivity.kt (hunk 8 above)
Comment (chuongdevJC) [inline app/src/main/java/com/hieuminh/chatapp/ui/MainActivity.kt]: remove if un-use else put it a comment

**Review by chuongdevJC — APPROVED** [review]

**Pull request merged**